kinesis/src/main/scala/akka/stream/alpakka/kinesis/scaladsl/KinesisSchedulerSource.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.kinesis.scaladsl
+
+import java.util.concurrent.Semaphore
+
+import akka.stream.Supervision.{Resume, Stop}
+import akka.stream._
+import akka.stream.scaladsl.{Flow, GraphDSL, Keep, Sink, Source, Zip}
+import akka.{Done, NotUsed}
+import akka.stream.alpakka.kinesis.Errors.{BackpressureTimeout, SchedulerUnexpectedShutdown}
+import akka.stream.alpakka.kinesis.{
+  CommittableRecord,
+  KinesisSchedulerCheckpointSettings,
+  KinesisSchedulerSourceSettings,
+  ShardProcessor
+}
+import software.amazon.kinesis.coordinator.Scheduler
+import software.amazon.kinesis.exceptions.ShutdownException
+import software.amazon.kinesis.processor.{ShardRecordProcessor, ShardRecordProcessorFactory}
+import software.amazon.kinesis.retrieval.KinesisClientRecord
+
+import scala.collection.immutable
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.control.Exception
+import scala.util.{Failure, Success}
+
+object KinesisSchedulerSource {
+
+  def apply(
+      schedulerBuilder: ShardRecordProcessorFactory => Scheduler,
+      settings: KinesisSchedulerSourceSettings = KinesisSchedulerSourceSettings.defaultInstance
+  )(implicit schedulerExecutor: ExecutionContext): Source[CommittableRecord, Scheduler] =
```

Comment:
A bit surprising to expect an execution context here, this would either need a full-blown `GraphStage` or the trick with the `SetupStage` we use in other Alpakka modules.

Reply:
Can you point me to an specific example?

Reply:
S3 makes heavy use of it. https://github.com/akka/alpakka/blob/master/s3/src/main/scala/akka/stream/alpakka/s3/impl/SetupStage.scala
It will soon be available in Akka Streams by default https://github.com/akka/akka/pull/26477